### PR TITLE
generate binary literals for enums that are already binary

### LINKED ISSUE #8440: Invoking generate member into enum with binary literals doesn't continue to use binary literals in both C# and VB
1. Copy the following into a VB project with binary literals active:
```
Module Module1
    Enum E
        b = &b01
    End Enum

    Sub Main()
        Console.WriteLine(E.c)
    End Sub
End Module
```
2. Go to E.c, 
3. Invoke "generate member"

*Expected:* we generate the new constant with binary literals, like we do for hex literals
*Actual:* we generate it with 2.

*Note:* this reproduces in both C# and VB.

(Found as a part of #8351)

### PR DESCRIPTION
**Pull request by TyOverby**
closes #8440

### CODE REVIEW

**Comment by TyOverby** [comment]
@dotnet/roslyn-compiler 

Please review when you have time!

**Comment by gafter** [comment]
Looks OK to me, but I think you need signoff from @dotnet/roslyn-ide

**Comment by jasonmalinowski** [comment]
:+1:

**Comment by CyrusNajmabadi** [comment]
can we have some tests around interesting values.  like the binary equiv of int.max.  or just if there's a binary value that exceeds all allowable values.  Thanks!

File: src/EditorFeatures/CSharpTest/Diagnostics/GenerateEnumMember/GenerateEnumMemberTests.cs [hunk 1]
```diff
@@ -130,6 +130,14 @@ public class GenerateEnumMemberTests : AbstractCSharpDiagnosticProviderBasedUser
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateEnumMember)]
+        public async Task TestBinary()
+        {
+            await TestAsync(
+@"class Program { void Main ( ) { Color . [|Blue|] ; } } enum Color { Red = 0b01 } ",
+@"class Program { void Main ( ) { Color . Blue ; } } enum Color { Red = 0b01 , Blue = 0b10 } ");
```
Comment (jcouv) [inline src/EditorFeatures/CSharpTest/Diagnostics/GenerateEnumMember/GenerateEnumMemberTests.cs]: What happens if `Red = 0b11`?
The code below seems to be hardcoded to substring of 2 characters, but I think binary literals allow more than 2 bits, right?

**Comment by TyOverby** [comment]
@CyrusNajmabadi Binary int-max test added.  It's just a copy-paste from the similar hex test.

**Comment by CyrusNajmabadi** [comment]
:+1:

**Pull request merged**